src/plugins/messageStorage/sqlite.js

```diff
@@ -188,9 +188,52 @@ class MessageStorage {
 		});
 	}
 
+	search(query) {
+		if (!this.isEnabled) {
+			return Promise.resolve([]);
+		}
+
+		let select =
+			'SELECT msg, type, time, channel FROM messages WHERE type = "message" AND network = ? AND json_extract(msg, "$.text") LIKE ?';
+		const params = [query.networkUuid, `%${query.searchTerm}%`];
+
+		if (query.channelName) {
+			select += " AND channel = ? ";
+			params.push(query.channelName);
+		}
+
+		select += " ORDER BY time DESC LIMIT 100 OFFSET ? ";
+		params.push(query.offset || 0);
```

Comment:
Add `query.offset = parseInt(query.offset, 10);`

Reply:
What is that for? offset should already be a number.

Reply:
It comes from socket, not guaranted to be.